src/com/sap/piper/Credential.groovy

```diff
@@ -0,0 +1,24 @@
+package com.sap.piper
+
+import com.cloudbees.groovy.cps.NonCPS
+
+class Credential implements Serializable {
+    static final long serialVersionUID = 1L
+
+    def alias
+    def username
+    def password
+
+    public Credential(alias, username, password) {
+        this.alias = alias
+        this.username = username
+        this.password = password
+    }
+
+    @NonCPS
+    def String toString() {
```

Comment:
This is actually used for the JSON-encoding functionality, correct? It doesn't feel right to use "toString()" for that. Also, no escaping is performed.

Reply:
Yeah you are right. Actually this whole way of handling the JSON is not required anymore (the related parts are already 3 years old and I forgot to review if its still relevant).
I rearranged the JSON handling with the latest commit.